### PR TITLE
Refactor code.json file generation

### PR DESCRIPTION
**Pull request by jgarber623-gov**
This PR is tangentially related to #231.

**TL;DR:** managing `code.json` just got a bit easier!

### Details and whatnot…

This pull request breaks apart the increasingly large `code.json` file into smaller, release-specific files and adds a Jekyll plugin to build a valid `code.json` from these files. The smaller, release-specific files are organized in `src/_releases/` and are organized in the form of `<repo domain>/<repo org/username>/<repo name>.json`.

For example, the code.mil repo's JSON file is located at `src/_releases/github.com/Code-dot-mil/code.mil.json`. Capitalization isn't required, but proper casing is encouraged to match the organization name (or username) and repository name.

Adding a new project/release to the `code.json` inventory file would now be a matter of creating the appropriate folder structure and JSON file in `src/_releases` with the relevant release-specific values from [version 2.0.0 of the `code.json` schema](https://code.gov/#/policy-guide/docs/compliance/inventory-code).

### CODE REVIEW

File: src/_plugins/code_gov_inventory_generator.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+      def page_content
+        output = {
+          agency: 'DOD',
+          version: '2.0.0',
+          measurementType: {
+            method: 'projects'
+          },
+          releases: []
+        }
```
Comment (jgarber623-gov) [inline src/_plugins/code_gov_inventory_generator.rb]: Standard top-level `code.json` bits here.

File: _config.yml [hunk 2]
```diff
@@ -21,6 +21,10 @@ defaults:
       path: "assets"
     values:
       layout: null
+  - scope:
+      path: "code.json"
+    values:
+      layout: null
```
Comment (jgarber623-gov) [inline _config.yml]: Otherwise `code.json` renders with the default HTML layout… 😭

**Comment by jordangov** [comment]
I'm wavering on the file structure... seems heavy. I talked to @jgarber623-gov about using a flat-file structure, but then we have to get into file name parsing pretty heavily. I guess my question would be: if someone is not on github, and their chosen platform doesn't do good URL path management, then what would we expect the outcome to be? That is... let's say I use ddsgit.com and my repo is at: https://www.ddsgit.com/orgs/team-america/jordan/awesome-project/view/index.html ... how would you expect that entry to look?
